src/lambda/commands/uploadLambda.ts

```diff
@@ -197,37 +197,40 @@ function createConfirmDeploymentPrompter(lambda: LambdaFunction) {
     })()
 }
 
-interface UploadLambdaWizardState {
+export interface UploadLambdaWizardState {
     readonly uploadType: 'zip' | 'directory'
     readonly targetUri: vscode.Uri
     readonly directoryBuildType: 'zip' | 'sam'
     readonly confirmedDeploy: boolean
     readonly lambda: LambdaFunction
 }
 
-class UploadLambdaWizard extends Wizard<UploadLambdaWizardState> {
+export class UploadLambdaWizard extends Wizard<UploadLambdaWizardState> {
     constructor(lambda?: LambdaFunction, invokePath?: vscode.Uri) {
         super({ initState: { lambda } })
         this.form.lambda.region.bindPrompter(() => createRegionPrompter().transform(region => region.id))
 
         this.form.uploadType.bindPrompter(() => createUploadTypePrompter())
-
-        this.form.targetUri.bindPrompter(({ uploadType }) => {
-            if (uploadType === 'directory') {
-                return createSingleFileDialog({
-                    canSelectFolders: true,
-                    canSelectFiles: false,
-                })
-            } else {
-                return createSingleFileDialog({
-                    canSelectFolders: false,
-                    canSelectFiles: true,
-                    filters: {
-                        'ZIP archive': ['zip'],
-                    },
-                })
-            }
-        })
+        if (invokePath && fs.statSync(invokePath.fsPath).isDirectory()) {
```

Comment:
Same behavior could also be achieved by conditionally initializing the state. This may be more readable though.